staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -107,6 +109,43 @@ func (mounter *Mounter) hasSystemd() bool {
 	return *mounter.withSystemd
 }
 
+// Do a bind mount including the needed remount for applying the bind opts.
+// If the remount fails figure out if the source filesystem has the ro, nodev,
+// noexec, nosuid, noatime, relatime or nodiratime flag set and try another
+// remount with the found flags.
+func (mounter *Mounter) BindMountSensitive(mounterPath string, mountCmd string, source string, target string, fstype string, bindOpts []string, bindRemountOpts []string, bindRemountOptsSensitive []string, mountFlags []string, systemdMountRequired bool) error {
```

Comment:
This should be probably done only when kubelet is running inside UserNS?

Reply:
Does it harm doing it when kubelet is not running inside UserNS? The first mount with the `bindRemountOpts` either succeeds initially without any error (because we are root in the host UserNS) or if the mount error is not fixable by adding the ro, nodev, noexec, nosuid, noatime, relatime or nodiratime flag we return the error.

Reply:
I would still keeping this feature-gated until we have had a chance of broader testing. Also why is this function exported (i.e starts with capital letter) ?

Reply:
The export topic is addressed now. Thanks for catching it. What kind of feature gating do you think of? Only apply if we run inside a UserNS? Is there any standard source of information inside Kubernetes that tells me that we run inside a UserNS or would it require writing some detection code?

Reply:
Sorry for a possible stupid beginners question: Do we need a new feature gate here or can we hook on an existing one? If a new one is needed is there a process for getting a new one or will it be done 'just' via an appropriate code change? If an existing one should be used can you propose an appropriate one that I should check for in the code?

Reply:
Bit of answering myself: Does the existing `KubeletInUserNamespace` sound acceptable for being used here?

Reply:
@gnufied : I now guarded the remount try with the feature gate KubeletInUserNamespace (e4c9d5aeefa54338d2bc074f0d12a0b24a729b81). This did not work fine initially as the needed import of `kubefeatures "k8s.io/kubernetes/pkg/features"` caused one of the CI jobs to break notifying an invalid dependency of `mount-utils` on `k8s.io/kubernetes`. I then replaced `kubefeatures.KubeletInUserNamespace` with the literal string `"KubeletInUserNamespace"` (210bacbbc9a1ab8ad463689ad4656403dc065eb2) which I understand is likely not the final solution. But it helped me run further tests. Is there a hint how I could get back to using `kubefeatures.KubeletInUserNamespace`again?

Reply:
Any hints, how I could use `kubefeatures.KubeletInUserNamespace`?

Reply:
@gnufied : Any hints, how I could use kubefeatures.KubeletInUserNamespace?

Reply:
> @AkihiroSuda : Any hints, how I could use kubefeatures.KubeletInUserNamespace?

This gate is enabled by tools that run Kubernetes inside Rootless Docker (or Rootless Podman):
- kind: https://kind.sigs.k8s.io/docs/user/rootless/
- minikube: https://minikube.sigs.k8s.io/docs/drivers/docker/ (See the `Rootless Docker` tab)
- Usernetes (similar to `kind` but multi-node): https://github.com/rootless-containers/usernetes

Reply:
@gnufied : Can you please check if this meets your requirements?

Reply:
@gnufied, @andyzhangx, @AkihiroSuda: Does this meet the requirements? Anything else I can do to move this forward?

Reply:
Looks good, but I'm not a maintainer here

Reply:
@gnufied, @andyzhangx : Any feedback?

Reply:
This has been addressed in the current PR.